src/nv_ingest/modules/sources/message_broker_task_source.py

```diff
@@ -0,0 +1,212 @@
+            control_message.add_task(task["type"], task["task_properties"])
+
+        # Debug Tracing
+        if do_trace_tagging:
+            ts_exit = datetime.now()
+            control_message.set_metadata("config::add_trace_tagging", do_trace_tagging)
+            control_message.set_timestamp(f"trace::entry::{MODULE_NAME}", ts_entry)
+            control_message.set_timestamp(f"trace::exit::{MODULE_NAME}", ts_exit)
+
+            if ts_send is not None:
+                control_message.set_timestamp("trace::entry::broker_source_network_in", ts_send)
+                control_message.set_timestamp("trace::exit::broker_source_network_in", ts_fetched)
+
+            if trace_id is not None:
+                # C++ layer in set_metadata errors out due to size of trace_id if it's an integer.
+                if isinstance(trace_id, int):
+                    trace_id = format_trace_id(trace_id)
+                control_message.set_metadata("trace_id", trace_id)
+
+            control_message.set_timestamp("latency::ts_send", datetime.now())
+    except Exception as e:
+        if "job_id" in job:
+            job_id = job["job_id"]
+            response_channel = f"{job_id}"
+            control_message.set_metadata("job_id", job_id)
+            control_message.set_metadata("response_channel", response_channel)
+            control_message.set_metadata("cm_failed", True)
+            annotate_cm(control_message, message="Failed to process job submission", error=str(e))
+        else:
+            raise
+
+    return control_message
+
+
+@register_module(MODULE_NAME, MODULE_NAMESPACE)
+def _message_broker_task_source(builder: mrc.Builder):
+    """
+    A module for receiving messages from a message broker, converting them into DataFrames,
+    and attaching job IDs to ControlMessages.
+
+    Parameters
+    ----------
+    builder : mrc.Builder
+        The Morpheus pipeline builder object.
+    """
+
+    validated_config = fetch_and_validate_module_config(builder, MessageBrokerTaskSourceSchema)
+
+    # Determine the client type and create the appropriate client
+    client_type = validated_config.broker_client.client_type.lower()
+    broker_params = validated_config.broker_client.broker_params or {}
+
+    if client_type == "redis":
+        client = RedisClient(
+            host=validated_config.broker_client.host,
+            port=validated_config.broker_client.port,
+            db=broker_params.get("db", 0),
+            max_retries=validated_config.broker_client.max_retries,
+            max_backoff=validated_config.broker_client.max_backoff,
+            connection_timeout=validated_config.broker_client.connection_timeout,
+            use_ssl=broker_params.get("use_ssl", False),
+        )
+    elif client_type == "simple":
+        # Start or retrieve the singleton SimpleMessageBroker server
+        # TODO(Devin) add config param for max_queue_size
+        max_queue_size = broker_params.get("max_queue_size", 10000)
```

Comment:
I think this answers my earlier comment about the queue being full. Seems like a max imposed size of 10000